### ISSUE TITLE
Issues with the internal queue

### ISSUE THREAD

**Issue by tf** [goldfire/howler.js#912]
We've been running into subtle bugs of the internal queuing system that is used to replay commands that have been invoked before the audio file finished loading.

To demonstrate the problems, I have setup some infrastructure for unit testing () and created some failing examples:

> https://github.com/tf/howler.js/compare/fade-eq-args...queue-issues ([Failing Travis build](https://travis-ci.org/tf/howler.js/builds/346880070))

* a call to `pause` is ignored while the player is waiting for the promise returned by `play`.

* the order of calls can change, when asyncronous commands (like `play` or `fade`) have been invoked before the load event. This can cause a Howl to end up with volume 0 even though the `volume(1)` call happened after the `volume(0)` call.

My proposed fix for which I can create a separate PR once # is merged is:

> https://github.com/tf/howler.js/compare/queue-issues...queued-commands ([Passing Travis build](https://travis-ci.org/tf/howler.js/builds/346880006))

The main idea is:

* Queue only one command per type. For example, there is no need to replay a full sequence of `play` and `pause` calls that happened before the load event. Basically the last command wins. Same is true for `volume` calls.

* Play queued commands synchronously in fixed order. This prevents queued commands from happening after commands that were invoked directly after the load event.

* Change the way `fade` works when it is called before the load event fired. Before, we replayed all queued fades one after the other. Once media has loaded, calls to `fade` cancel previous fades anyway. The change makes this behavior more consistent. In particular, drop queued fade commands when volume is queued.

**Comment by tf** [comment]
@goldfire Any chance my proposed fix will be considered/this issue will be addressed? We would really like to use Howler in our project, but with these bugs, I really don't like the prospect of living on a forked version forever.
